bindings_ffi/src/mls.rs

```diff
@@ -2294,6 +2294,76 @@ mod tests {
         assert_eq!(bo_messages5.len(), 1);
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 5)]
+    async fn test_can_sync_all_groups_active_only() {
+        let alix = new_test_client().await;
+        let bo = new_test_client().await;
+
+        // Create 30 groups with alix and bo and sync them
+        for _i in 0..30 {
+            alix.conversations()
+                .create_group(
+                    vec![bo.account_address.clone()],
+                    FfiCreateGroupOptions::default(),
+                )
+                .await
+                .unwrap();
+        }
+        bo.conversations().sync().await.unwrap();
+        bo.conversations().sync_all_groups().await.unwrap();
+
+        // Alix sends a message in each group
+        for group in alix
+            .conversations()
+            .list(FfiListConversationsOptions::default())
+            .await
+            .unwrap()
+        {
+            group.send("hello".as_bytes().to_vec()).await.unwrap();
+        }
+
+        // Get duration to sync all groups
+        let start_time_1 = std::time::Instant::now();
+        bo.conversations().sync_all_groups().await.unwrap();
+        let duration_1 = start_time_1.elapsed();
+
+        // Remove bo from all groups and sync
+        for group in alix
+            .conversations()
+            .list(FfiListConversationsOptions::default())
+            .await
+            .unwrap()
+        {
+            group
+                .remove_members(vec![bo.account_address.clone()])
+                .await
+                .unwrap();
+        }
+        bo.conversations().sync_all_groups().await.unwrap();
+
+        // Alix sends one more message in each group
+        for group in alix
+            .conversations()
+            .list(FfiListConversationsOptions::default())
+            .await
+            .unwrap()
+        {
+            group.send("hello".as_bytes().to_vec()).await.unwrap();
+        }
+
+        // Verify that sync all groups is faster after bo is removed
+        let start_time_2 = std::time::Instant::now();
+        bo.conversations().sync_all_groups().await.unwrap();
+        let duration_2 = start_time_2.elapsed();
```

Comment:
I guess you have to do it this way because sync_all_groups returns (). 

What if we just made it return the number of groups synced? Feels like that would be useful beyond just test cases.

Reply:
good call - added here https://github.com/xmtp/libxmtp/pull/992/commits/856ac68775c9087f21dbb7f9bf407e723840da5c